devices/webcam/hand_tracking_utility.py

```diff
@@ -0,0 +1,600 @@
+mp_hands: HandsType = mp.solutions.hands
+
+parser = ap.ArgumentParser()
+parser.add_argument("-te", "--test", help = "enable test mode", action = "store_true", required = False)
+parser.add_argument("-li", "--log-images", help = "add image data to logs", action = "store_true", required = False)
+parser.add_argument("-lp", "--log-poses", help = "add pose estimation data to logs", action = "store_true", required = False)
+parser.add_argument("-lf", "--log-format", type = str, nargs = 1, choices = ["hdf5"], help = "sepcify logging format. Supports hdf5, default is hdf5", action = "store", required = False)
+parser.add_argument("-ld", "--log-dir", type = str, nargs = 1, help = "sepcify log directory (must be full path), default is ./logs", action = "store", required = False)
+parser.add_argument("-ln", "--log-name", type = str, nargs = 1, help = "sepcify log filename, default is hand_tracking_utility", action = "store", required = False)
+parser.add_argument("-rp", "--replay", help = "replay a log file", action = "store_true", required = False)
+parser.add_argument("-dn", "--device-number", type = int, nargs = 1, help = "specify device number to capture from, default is 0", action = "store", required = False)
+parser.add_argument("-sr", "--sample-rate", type = int, nargs = 1, help = "specify sample rate from device in Hz, default is 30", action = "store", required = False)
+parser.add_argument("-dr", "--device-resolution", type = int, nargs = 2, help = "specify resolution for capture device, default is 854x480", action = "store", required = False)
+parser.add_argument("-ri", "--rescale-image", help = "enable rescaling display window from input size", action = "store_true", required = False)
+parser.add_argument("-rr", "--rescale_resolution", type = int, nargs = 2, help = "specify resolution to rescale display stream to, default is 1280x720", action = "store", required = False)
+
+class PoseFrame(lg.Message):
+    """
+    Represents a single frame of a webcam's stream along with its hand tracking estimations
+
+    @attributes:
+        frame: a (W, H, 3) shape array of the webcam's stream at its default resolution, in the blue/green/red color space
+        landmarks: List[NormalizedLandmarkList], one NormalizedLandmarkList object for each detected hand
+    
+    NormalizedLandmarkList:
+        @attributes:
+            landmark: RepeatedCompositeContainer, list-like container with up to 21 hand pose landmarks with their normalized (X, Y, Z) coordinates as attributes
+    """
+
+    frame: np.ndarray
+    landmarks: List[NormalizedLandmarkList]
+
+class PoseEstimationConfig(lg.Config):
+    """
+    Configuration for MediaPipe's hand pose esitmation graph
+    See https://google.github.io/mediapipe/solutions/hands.html
+
+    @attributes:
+        model_complexity: int
+        min_detection_confidence: float
+        min_tracking_confidence: float
+    """
+
+    model_complexity: int
```

Comment:
Default values be added in this config based on https://google.github.io/mediapipe/solutions/hands.html